### PR TITLE
[GlusterVolStatus] Fix process name for longer host or brick name

### PR DESCRIPTION
**Pull request by psachin**
Resolves: #2232

Signed-off-by: Sachin Patil <psachin@redhat.com>

### CODE REVIEW

**Comment by psachin** [comment]
@SonaArora This should fix the issue. Please check.

File: insights/parsers/gluster_vol.py [hunk 1]
```diff
@@ -130,20 +130,29 @@ class having a `data` attribute.
     def parse_content(self, content):
         # Stored data in a dictionary data structure
         self.data = {}
-        content = get_active_lines(content, '----')
+
+        # Needed for content with multiple tables
+        content = [line for line in get_active_lines(content, '----') if not line.startswith('There are no active volume tasks')]
```
Comment (xiangce) [inline insights/parsers/gluster_vol.py]: If you just want to ignore this line, a better way is to add it to the `trailing_ingore` list of the `parse_fixed_table`

File: insights/parsers/gluster_vol.py [hunk 2]
```diff
@@ -130,20 +130,29 @@ class having a `data` attribute.
...
+                    ('Gluster process', 'Gluster_process'),
+                    ('TCP Port', 'TCP_Port'),
+                    ('RDMA Port', 'RDMA_Port')
+                ],
+                heading_ignore=['Gluster process'],
+                trailing_ignore=['Task Status of Volume'])
+
+            # For process names having longer hostnames
+            for n, i in enumerate(body):
+                if i['TCP_Port'] == '' and i['RDMA_Port'] == '' and i['Online'] == '' and i['Pid'] == '':
```
Comment (xiangce) [inline insights/parsers/gluster_vol.py]: A pythonic way is:

`if i['TCP_Port'] == i['RDMA_Port'] == i['Online'] == i['Pid'] == '':`

File: insights/parsers/gluster_vol.py (hunk 1 above)
Comment (psachin) [reply insights/parsers/gluster_vol.py]: Sure.

File: insights/parsers/gluster_vol.py (hunk 2 above)
Comment (psachin) [reply insights/parsers/gluster_vol.py]: Done. Please check now.

**Comment by SonaArora** [comment]
@psachin Works well.

Tested against the below data:

Status of volume: test
Gluster process                             TCP Port  RDMA Port  Online  Pid
------------------------------------------------------------------------------
Brick Amyblackberry01.ocp.examplesyst.com:/
exports/gluster/ber/brick1                  49153     0          Y       9011
Brick Amyblackberry02.ocp.examplesyst.com:/
exports/gluster/ber/brick1                  49153     0          Y       4246
Brick Amyblackberry03.ocp.examplesyst.com:/
exports/gluster/ber/brick1                  49153     0          Y       40939
Self-heal Daemon on localhost               N/A       N/A        Y       6000
Self-heal Daemon on Amyblackberry02.ocp.exa
mplesyst.com                                N/A       N/A        Y       70612
Self-heal Daemon on Amyblackberry03.ocp.exa
mplesyst.com                                N/A       N/A        Y       71934

Task Status of Volume test
------------------------------------------------------------------------------
There are no active volume tasks


The output from the parser:
(Pdb) print(body)
[{'Gluster_process': 'Brick Amyblackberry01.ocp.examplesyst.com:/exports/gluster/ber/brick1', 'TCP_Port': '49153', 'RDMA_Port': '0', 'Online': 'Y', 'Pid': '9011'}, {'Gluster_process': 'Brick Amyblackberry02.ocp.examplesyst.com:/exports/gluster/ber/brick1', 'TCP_Port': '49153', 'RDMA_Port': '0', 'Online': 'Y', 'Pid': '4246'}, {'Gluster_process': 'Brick Amyblackberry03.ocp.examplesyst.com:/exports/gluster/ber/brick1', 'TCP_Port': '49153', 'RDMA_Port': '0', 'Online': 'Y', 'Pid': '40939'}, {'Gluster_process': 'Self-heal Daemon on localhost', 'TCP_Port': 'N/A', 'RDMA_Port': 'N/A', 'Online': 'Y', 'Pid': '6000'}, 
{'Gluster_process': 'Self-heal Daemon on Amyblackberry02.ocp.examplesyst.com', 'TCP_Port': 'N/A', 'RDMA_Port': 'N/A', 'Online': 'Y', 'Pid': '70612'}, 
{'Gluster_process': 'Self-heal Daemon on Amyblackberry03.ocp.examplesyst.com', 'TCP_Port': 'N/A', 'RDMA_Port': 'N/A', 'Online': 'Y', 'Pid': '71934'}]

Thanks a lot.

**Pull request merged**